### ISSUE TITLE
Scan output files should indicate which facets/presets were run

### ISSUE THREAD

**Issue by mvogelgesang** [GSA/EDX#756]
## Summary

When running scans, the resulting output files will often have blank or default values. This can create confusion when trying to determine if the data is factual or a default. Instead, update the output so that:

- [x] Only requested facets are included as JSON output
- [x] Facets/ Presets are listed in the first level of keys 
- [x] Add error key(s) to each facet where scan errors can be logged. 
  - To do this, the functions currently referenced in `helper/scan.ts` need to be converted to classes which can implement a standard interface. The interface has a single async method called `run()` which returns an object which contains `data` and `error` attributes.

## JSON Output Organization

Current JSON output is inconsistent when it comes to reporting facet results. Some keys are objects of results while others are objects containing description of the scan and data stored either in an array or object. Since keys reside on the same level as scan metadata, it can feel messy.

```json
"performanceMetric": {
    "description": "Represents GSA IT FY22 performance metrics",
    "hsts": true,
    "dap": false,
    "contact": true,
    "banner": false,
    "identifier": false,
    "identifierAccessibility": false,
    "identifierFOIA": false,
    "identifierPrivacy": false,
    "search": false
  },
  "cuiBanner": {
    "description": "Looks for the CUI warning banner on site and returns a fuzzy match ranking between 0 and 1 where a higher score indicates a closer match. Two variants of the CUI warning banner exist and scores are provided for both the internal and external variants. Specific language can be found in source and CIO 2100.1M, https://www.gsa.gov/cdnstatic/CIO_21001M_GSA_Information_Technology_(IT)_Security_Policy_03-26-2021_CC044215.pdf",
    "data": []
  }
```

Updating to a structure which clearly separates facet reports from metadata while also following a consistent pattern would make for easier consumption.

```diff
 {
   "performanceMetric": {
     "description": "Represents GSA IT FY22 performance metrics",
-    "hsts": true,
-    "dap": false,
-    "contact": true,
-    "banner": false,
-    "identifier": false,
-    "identifierAccessibility": false,
-    "identifierFOIA": false,
-    "identifierPrivacy": false,
-    "search": false
+    "data": {
+      "hsts": true,
+      "dap": false,
+      "contact": true,
+      "banner": false,
+      "identifier": false,
+      "identifierAccessibility": false,
+      "identifierFOIA": false,
+      "identifierPrivacy": false,
+      "search": false
+    }
   },
   "cuiBanner": {
     "description": "Looks for the CUI warning banner on site and returns a fuzzy match ranking between 0 and 1 where a higher score indicates a closer match. Two variants of the CUI warning banner exist and scores are provided for both the internal and external variants. Specific language can be found in source and CIO 2100.1M, https://www.gsa.gov/cdnstatic/CIO_21001M_GSA_Information_Technology_(IT)_Security_Policy_03-26-2021_CC044215.pdf",
}

```

- [x] Change websiteReport to a class
- [x] Add method to websiteReport which allows additional facet reports to be added to it - this should take care of the desire for reports to only contain facets that were executed
- [x] Update screenshot and search engine facets to append data to existing screenshot section of report

## v1.0.0

This work may require substantial changes to the organization of JSON file outputs. As a result, consideration needs to be given to `data condense` and other functions which compile the outputs. What needs to be built/ changed to ensure backward compatibility? Should old files get automated structure updates or should `data condense` change to support both <1.0.0 and >1.0.0?

**Comment by mvogelgesang** [comment]
A handful of domains (usdaadvantage.gsa.gov) are responsible for errors being thrown, especially when running Lighthouse, both in the browser and with edxcli. Digging into this further, the following in-browser settings enabled Lighthouse to run:

```
Legacy Navigation: check
Clear storage: unchecked
Simulated throttling: default
```

Will try setting these values in the cli to see if that produces better results.

**Comment by mvogelgesang** [comment]
Merged into #769 feature branch
